src/frontend/src/binder/relation/join.rs

```diff
@@ -141,7 +141,7 @@ impl Binder {
                     .context
                     .indices_of
                     .iter()
-                    .filter(|(s, _)| *s != "_row_id") // filter out `_row_id`
```

Comment:
So the name is `t._row_id` in this test case and was not filtered out as expected. Do we have other places checking for `_row_id` instead of `is_hidden`?